server/test/unit/common/config/test_external_config.py

```diff
@@ -0,0 +1,107 @@
+
+
+class TestExternalConfig(unittest.TestCase):
+    def test_type_convert(self):
+        # The values from environment variables and aws secrets are returned as strings.
+        # These values need to be converted to the proper types.
+
+        self.assertEqual(ExternalConfig.convert_value("1"), int(1))
+        self.assertEqual(ExternalConfig.convert_value("1.1"), float(1.1))
+        self.assertEqual(ExternalConfig.convert_value("string"), "string")
+        self.assertEqual(ExternalConfig.convert_value("true"), True)
+        self.assertEqual(ExternalConfig.convert_value("True"), True)
+        self.assertEqual(ExternalConfig.convert_value("false"), False)
+        self.assertEqual(ExternalConfig.convert_value("False"), False)
+        self.assertEqual(ExternalConfig.convert_value("False"), False)
+        self.assertEqual(ExternalConfig.convert_value("null"), None)
+        self.assertEqual(ExternalConfig.convert_value("None"), None)
+        self.assertEqual(ExternalConfig.convert_value("{'a':10, 'b':'string'}"), dict(a=int(10), b="string"))
+
+    def test_environment_variable(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            configfile = os.path.join(tempdir, "config.yaml")
+            with open(configfile, "w") as fconfig:
+                config = """
+                external:
+                  environment:
+                    - name: DATAPATH
+                      path: [server, single_dataset, datapath]
+                      required: true
+                    - name: DIFFEXP
+                      path: [dataset, diffexp, enable]
+                      required: true
+                """
+                fconfig.write(config)
+
+            env = os.environ
+            env["DATAPATH"] = f"{FIXTURES_ROOT}/pbmc3k.cxg"
+            env["DIFFEXP"] = "False"
+            with test_server(command_line_args=["-c", configfile], env=env) as server:
+                session = requests.Session()
+                response = session.get(f"{server}/api/v0.2/config")
+                data_config = response.json()
+                self.assertEqual(data_config["config"]["displayNames"]["dataset"], "pbmc3k")
+                self.assertTrue(data_config["config"]["parameters"]["disable-diffexp"])
+
+            env["DATAPATH"] = f"{FIXTURES_ROOT}/a95c59b4-7f5d-4b80-ad53-a694834ca18b.h5ad"
+            env["DIFFEXP"] = "True"
+            with test_server(command_line_args=["-c", configfile], env=env) as server:
+                session = requests.Session()
+                response = session.get(f"{server}/api/v0.2/config")
+                data_config = response.json()
+                self.assertEqual(
+                    data_config["config"]["displayNames"]["dataset"], "a95c59b4-7f5d-4b80-ad53-a694834ca18b"
+                )
+                self.assertFalse(data_config["config"]["parameters"]["disable-diffexp"])
+
+    @patch("server.common.config.external_config.get_secret_key")
```

Comment:
An option - you could setup mocked AWS secrets manager using moto.

Reply:
I just looked into it.  It's a bit more overhead than what I would need it for here.   I'd have to create the secrets using moto first, then access the secrets.   Will keep it in mind for the future.